crates/client/mempool/src/block_production.rs

```diff
@@ -0,0 +1,159 @@
+                    range_check: config_bouncer.builtin_count.range_check / frac,
+                    range_check96: config_bouncer.builtin_count.range_check96 / frac,
+                },
+                gas: config_bouncer.gas / frac,
+                message_segment_length: config_bouncer.message_segment_length / frac,
+                n_events: config_bouncer.n_events / frac,
+                n_steps: config_bouncer.n_steps / frac,
+                state_diff_size: config_bouncer.state_diff_size / frac,
+            };
+        }
+
+        let mut txs_to_process = Vec::with_capacity(TX_CHUNK_SIZE);
+        self.mempool.take_txs_chunk(&mut txs_to_process, TX_CHUNK_SIZE);
+
+        let blockifier_txs: Vec<_> =
+            txs_to_process.iter().map(|tx| Transaction::AccountTransaction(clone_account_tx(&tx.tx))).collect();
+
+        // Execute the transactions.
+        let all_results = self.executor.execute_txs(&blockifier_txs);
+
+        // Split the `txs_to_process` vec into two iterators.
+        let mut to_process_iter = txs_to_process.into_iter();
+        // This iterator will consume the first part of `to_process_iter`.
+        let consumed_txs_to_process = to_process_iter.by_ref().take(all_results.len());
+
+        for (exec_result, mempool_tx) in Iterator::zip(all_results.into_iter(), consumed_txs_to_process) {
+            match exec_result {
+                Ok(execution_info) => {
+                    // Note: reverted txs also appear as Ok here.
+                    log::debug!("Successful execution of transaction {:?}", mempool_tx.tx_hash());
+
+                    // exec
+                }
+                Err(err) => {
+                    // Internal Server Error: this silently drops the transaction.
+                    log::error!("Unsuccessful execution of transaction {:?}: {err:#}", mempool_tx.tx_hash());
+                }
+            }
+        }
+
+        // This contains the rest of `to_process_iter`.
+        let rest_txs_to_process: Vec<_> = to_process_iter.collect();
+
+        // Add back the unexecuted transactions to the mempool.
+        self.mempool.readd_txs(rest_txs_to_process);
+
+        Ok(())
+    }
+
+    fn produce_block(&mut self) -> Result<(), Error> {
+        // Complete the block with full bouncer capacity.
+        let finish_block = true;
+        self.update_pending_block(finish_block)?;
+
+        // Convert the pending block to a closed block and save to db.
+
+        // Prepare for next block.
+        let mut pending_block = self.mempool.get_or_create_pending_block()?;
+        let bouncer_config = self.backend.chain_config().bouncer_config.clone();
+        self.executor = ExecutionContext::new(Arc::clone(&self.backend), &pending_block.info)?.tx_executor();
+        self.current_pending_tick = 0;
+
+        Ok(())
+    }
+
+    pub async fn block_production_task(&mut self) -> Result<(), anyhow::Error> {
```

Comment:
SHouldn't the signature be:
`pub async fn block_production_task(&mut self) -> !`

Reply:
I think I don't bubble up anything here right now but I might want to in the future
Could be Result<!, anyhow::Error> true but that doesnt really matter

I think all my services have that `Result<(), anyhow::Error>` type in the codebase